### PR TITLE
Fix mouse region crash when using closures

### PR DESCRIPTION
**Pull request by dkwingsmt**
## Description
This PR fixes an issue where `MouseRegion` crashes when being passed with closures instead of methods. 

It changes how a `RenderMouseRegion` handles its `MouseTrackingAnnotation`.
- Instead of creating a new annotation every time it becomes active and destroys it when deactivated, it now creates an annotation during the constructor and holds onto it until the end of its lifecycle.
- Instead of directly passing the argument callbacks to the annotation, it proxies them using methods.

### Regarding changed tests
`page_test.dart`: These two lines were changed from 2 to 3 a while ago when hover was introduced. https://github.com/flutter/flutter/pull/31438/files#r283037816
This PR happens to change them back to 2, which is good.

## Related Issues

- Might fix (but not sure) https://github.com/flutter/flutter/issues/35808 

## Tests

I added the following tests:
- MouseRegion should detect pointer enter with closure arguments

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 1]
```diff
@@ -2629,53 +2639,26 @@ class RenderMouseRegion extends RenderProxyBox {
   MouseTrackerAnnotation get hoverAnnotation => _hoverAnnotation;
 
   void _updateAnnotations() {
```
Comment (gspencergoog) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: Nice!  Much simpler.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: packages/flutter/test/material/page_test.dart [hunk 2]
```diff
@@ -297,7 +297,7 @@ void main() {
       ),
     );
     await tester.tap(find.text('PUSH'));
-    expect(await tester.pumpAndSettle(const Duration(minutes: 1)), 3);
+    expect(await tester.pumpAndSettle(const Duration(minutes: 1)), 2);
```
Comment (dkwingsmt) [inline packages/flutter/test/material/page_test.dart]: These two lines were changed from 2 to 3 a while ago when hover was introduced. https://github.com/flutter/flutter/pull/31438/files#r283037816
This PR happens to change them back to 2, which is good.

**Comment by dkwingsmt** [comment]
Current state: I'm waiting for 5bb8d8f to be rolled into google3 before merging dependent PRs such as this in case of a rollback.

**Comment by goderbauer** [comment]
@dkwingsmt Looks like the commit 5bb8d8f has been rolled into google3. Should we submit this PR?

**Comment by dkwingsmt** [comment]
This should also have fixed https://github.com/flutter/flutter/issues/36350

**Pull request merged**